### PR TITLE
BUG: fix ICA resetting

### LINKED ISSUE #2087: max_pca_components no reset during preprocessing.ica.ICA._reset
Hi everyone,

I've found that when refitting an ICA object, the property max_pca_components is not set to its original values during reset. This may lead to unexpected behaviour. 

Example:

+ Set "max_pca_components=None", so that the object automatically sets it to use all computed PCs through the line "self.max_pca_components = len(picks)" when fitted with a dataset. 

+ Now refit the object with a dataset with different number of sensors. As max_pca_components is not reset to None, the object won't be configured to use all the computed PCs (the previous value is kept), producing unexpected behaviour.

Thank you
Marcos

PS: I'm new on github so I'm not sure how this works. Should I fix this issue by myself and request a pull?

### PR DESCRIPTION
**Pull request by dengemann**
Addresses and closes #2087 .
Thank @mdelpozobanos for reporting.

### CODE REVIEW

File: mne/preprocessing/tests/test_ica.py [hunk 1]
```diff
@@ -128,6 +128,32 @@ def test_ica_rank_reduction():
         assert_true(n_components < n_pca_components <= rank_after <=
                     rank_before)
 
+@requires_sklearn
+def test_ica_reset():
+    """Test ICA resetting"""
+    raw = io.Raw(raw_fname, preload=True).crop(0, stop, False).crop(0.5)
```
Comment (agramfort) [inline mne/preprocessing/tests/test_ica.py]: why do you need to crop twice?

**Comment by agramfort** [comment]
travis is not happy

**Comment by dengemann** [comment]
pep8 issue fixed.

File: mne/preprocessing/tests/test_ica.py (hunk 1 above)
Comment (dengemann) [inline mne/preprocessing/tests/test_ica.py]: Copied from elsewhere, works. So let's keep it as is.

2015-05-13 13:57 GMT+02:00 Alexandre Gramfort <notifications@github.com>:

File: mne/preprocessing/tests/test_ica.py [hunk 2]
```diff
@@ -130,6 +130,33 @@ def test_ica_rank_reduction():
 
 
 @requires_sklearn
+def test_ica_reset():
+    """Test ICA resetting"""
+    raw = io.Raw(raw_fname, preload=True).crop(0, stop, False).crop(0.5)
```
Comment (Eric89GXL) [inline mne/preprocessing/tests/test_ica.py]: If we're going to keep this as is, can you make a note (`XXX`) and wherever you copied it from that we should investigate at some point why it matters? It is strange as it should be equivalent to `.crop(0.5, stop, False)`, and I'm worried if it has to be done this way there is some bug we're unaware of...

Comment (agramfort) [inline mne/preprocessing/tests/test_ica.py]: I don't like it either.

@Eric89GXL can you take over if @dengemann is busy?

thx

Comment (Eric89GXL) [inline mne/preprocessing/tests/test_ica.py]: Done -- PR opened into your branch @dengemann. The changes even speed up tests by ~20% by not preloading extra data and using `.preload_data()` after cropping (hooray!).

Comment (dengemann) [inline mne/preprocessing/tests/test_ica.py]: Amazing!!!!

2015-05-13 18:18 GMT+02:00 Eric Larson <notifications@github.com>:

**Comment by Eric89GXL** [comment]
good to go for you @dengemann ?

**Comment by dengemann** [comment]
Yes, good.

**Comment by Eric89GXL** [comment]
Thanks @dengemann!

**Pull request merged**